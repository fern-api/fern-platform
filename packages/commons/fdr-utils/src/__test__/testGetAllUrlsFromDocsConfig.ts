import { DocsV2Read } from "@fern-api/fdr-sdk";
import fs from "fs";
import path from "path";
import { getAllUrlsFromDocsConfig } from "../getAllUrlsFromDocsConfig";

export function testGetAllUrlsFromDocsConfig(fixtureName: string): void {
    // eslint-disable-next-line vitest/valid-title
    describe(fixtureName, () => {
        it("gets all urls from docs config", async () => {
            const fixturePath = path.join(__dirname, "fixtures", `${fixtureName}.json`);

            const content = fs.readFileSync(fixturePath, "utf-8");

            const fixture = JSON.parse(content) as DocsV2Read.LoadDocsForUrlResponse;

            const urls = getAllUrlsFromDocsConfig(
                fixture.baseUrl.domain,
                fixture.baseUrl.basePath,
                fixture.definition.config.navigation,
                fixture.definition.apis,
            );

            expect(urls).toMatchSnapshot();
        });
    });
}
