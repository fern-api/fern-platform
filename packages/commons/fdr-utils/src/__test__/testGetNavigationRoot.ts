import { DocsV2Read } from "@fern-api/fdr-sdk";
import fs from "fs";
import path from "path";
import { getNavigationRoot } from "../getNavigationRoot";

export function testGetNavigationRoot(fixtureName: string, slug: string): void {
    // eslint-disable-next-line vitest/valid-title
    describe(fixtureName, () => {
        it("gets navigation root for /" + slug, async () => {
            const fixturePath = path.join(__dirname, "fixtures", `${fixtureName}.json`);

            const content = fs.readFileSync(fixturePath, "utf-8");

            const fixture = JSON.parse(content) as DocsV2Read.LoadDocsForUrlResponse;

            const urls = getNavigationRoot(
                slug.split("/"),
                fixture.baseUrl.basePath,
                fixture.definition.apis,
                fixture.definition.config.navigation,
            );

            expect(stripForSnapshot(urls)).toMatchSnapshot();
        });
    });
}

function stripForSnapshot(obj: ReturnType<typeof getNavigationRoot>): unknown {
    if (obj == null || obj.type === "redirect") {
        return obj;
    }

    // only test the tab index, version index, and current node
    return {
        type: obj.type,
        found: {
            currentTabIndex: obj.found.currentTabIndex,
            currentVersionIndex: obj.found.currentVersionIndex,
            currentNode: obj.found.currentNode,
        },
    };
}
