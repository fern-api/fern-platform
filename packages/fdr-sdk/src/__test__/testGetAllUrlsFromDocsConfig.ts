import urljoin from "url-join";
import { FernNavigation } from "..";
import { NodeCollector } from "../navigation/NodeCollector";

export function testGetAllUrlsFromDocsConfig(
  root: FernNavigation.RootNode,
  domain: string
): void {
  it("gets all urls from docs config", async () => {
    const collector = NodeCollector.collect(root);
    const urls = collector.staticPageSlugs.map((slug) => urljoin(domain, slug));
    expect(urls).toMatchSnapshot();
  });
}
