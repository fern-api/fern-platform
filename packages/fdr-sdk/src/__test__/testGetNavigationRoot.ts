import { FernNavigation } from "..";
import { findNode } from "../navigation/utils";

export function testGetNavigationRoot(root: FernNavigation.RootNode, slug: string): void {
    it("gets navigation root for /" + slug, async () => {
        const found = findNode(root, FernNavigation.slugjoin(slug));

        if (found.type === "found") {
            expect(found.node).toMatchSnapshot();
            expect(found.currentVersion?.versionId).toMatchSnapshot();
            expect(found.currentTab?.slug).toMatchSnapshot();
        } else {
            expect(found).toMatchSnapshot();
        }
    });
}
