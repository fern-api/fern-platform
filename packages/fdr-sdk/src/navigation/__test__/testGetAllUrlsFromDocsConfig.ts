import urljoin from "url-join";
import { FernNavigation } from "../..";
import { NodeCollector } from "../NodeCollector";

export function testGetAllUrlsFromDocsConfig(root: FernNavigation.RootNode, domain: string): void {
    it("gets all urls from docs config", async () => {
        const slugCollector = NodeCollector.collect(root);
        const urls = slugCollector.getPageSlugs().map((slug) => urljoin(domain, slug));
        expect(urls).toMatchSnapshot();
    });
}
