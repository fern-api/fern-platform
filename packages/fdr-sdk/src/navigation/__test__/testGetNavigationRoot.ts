import fs from "fs";
import path from "path";
import { DocsV2Read } from "../../client";
import { convertLoadDocsForUrlResponse, findNode, slugjoin } from "../utils";

export function testGetNavigationRoot(fixtureName: string, slug: string): void {
    // eslint-disable-next-line vitest/valid-title
    describe(fixtureName, () => {
        it("gets navigation root for /" + slug, async () => {
            const fixturePath = path.join(__dirname, "fixtures", `${fixtureName}.json`);

            const content = fs.readFileSync(fixturePath, "utf-8");

            const fixture = JSON.parse(content) as DocsV2Read.LoadDocsForUrlResponse;
            const node = convertLoadDocsForUrlResponse(fixture);
            const found = findNode(node, slugjoin(slug));

            if (found.type === "found") {
                expect(found.node).toMatchSnapshot();
                expect(found.currentVersion?.versionId).toMatchSnapshot();
                expect(found.currentTab?.slug).toMatchSnapshot();
            } else {
                expect(found).toMatchSnapshot();
            }
        });
    });
}
