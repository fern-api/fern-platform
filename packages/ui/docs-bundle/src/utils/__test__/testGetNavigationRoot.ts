import { DocsV2Read } from "@fern-api/fdr-sdk";
import fs from "fs";
import path from "path";
import { getNavigationRoot } from "../getNavigationRoot";

// eslint-disable-next-line jest/no-export
export function testGetNavigationRoot(fixtureName: string, slug: string): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("gets navigation root for " + slug, async () => {
            const fixturePath = path.join(__dirname, "fixtures", `${fixtureName}.json`);

            const content = fs.readFileSync(fixturePath, "utf-8");

            const fixture = JSON.parse(content) as DocsV2Read.LoadDocsForUrlResponse;

            const urls = getNavigationRoot(
                slug.split("/"),
                fixture.baseUrl.basePath,
                fixture.definition.apis,
                fixture.definition.config.navigation,
            );

            expect(urls).toMatchSnapshot();
        });
    });
}
