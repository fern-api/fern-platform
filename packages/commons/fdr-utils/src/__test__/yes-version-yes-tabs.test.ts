import { testGetAllUrlsFromDocsConfig } from "./testGetAllUrlsFromDocsConfig";
import { testGetNavigationRoot } from "./testGetNavigationRoot";

const FIXTURE = "yes-version-yes-tabs";

describe("getAllUrlsFromDocsConfig", () => {
    testGetAllUrlsFromDocsConfig(FIXTURE);
});

describe("getNavigationRoot", () => {
    testGetNavigationRoot(FIXTURE, "docs/api");
    testGetNavigationRoot(FIXTURE, "docs/api/tab-1");
    testGetNavigationRoot(FIXTURE, "docs/api/tab-2");
    testGetNavigationRoot(FIXTURE, "docs/api/version-1/section-2");
    testGetNavigationRoot(FIXTURE, "docs/api/version-1");
    testGetNavigationRoot(FIXTURE, "docs/api/version-2");
    testGetNavigationRoot(FIXTURE, "docs/api/version-2/tab-2");
    testGetNavigationRoot(FIXTURE, "docs/api/version-2/tab-2/section-5");
});
