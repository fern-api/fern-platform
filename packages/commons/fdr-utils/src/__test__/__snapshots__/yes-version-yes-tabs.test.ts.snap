// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAllUrlsFromDocsConfig yes-version-yes-tabs gets all urls from docs config 1`] = `
[
  "yes-version-yes-tabs/docs/api",
  "yes-version-yes-tabs/docs/api/tab-1",
  "yes-version-yes-tabs/docs/api/tab-1/section-1",
  "yes-version-yes-tabs/docs/api/tab-1/section-1/page-1",
  "yes-version-yes-tabs/docs/api/page-2",
  "yes-version-yes-tabs/docs/api/tab-1/section-2",
  "yes-version-yes-tabs/docs/api/section-2",
  "yes-version-yes-tabs/docs/api/tab-1/section-2/page-4",
  "yes-version-yes-tabs/docs/api/tab-1/page-5",
  "yes-version-yes-tabs/docs/api/tab-2",
  "yes-version-yes-tabs/docs/api/tab-2/page-7",
  "yes-version-yes-tabs/docs/api/tab-2/page-8",
  "yes-version-yes-tabs/docs/api/tab-2/section-5",
  "yes-version-yes-tabs/docs/api/tab-2/section-5/page-10",
  "yes-version-yes-tabs/docs/api/version-1",
  "yes-version-yes-tabs/docs/api/version-1/tab-1",
  "yes-version-yes-tabs/docs/api/version-1/tab-1/section-1",
  "yes-version-yes-tabs/docs/api/version-1/tab-1/section-1/page-1",
  "yes-version-yes-tabs/docs/api/version-1/page-2",
  "yes-version-yes-tabs/docs/api/version-1/tab-1/section-2",
  "yes-version-yes-tabs/docs/api/version-1/section-2",
  "yes-version-yes-tabs/docs/api/version-1/tab-1/section-2/page-4",
  "yes-version-yes-tabs/docs/api/version-1/tab-1/page-5",
  "yes-version-yes-tabs/docs/api/version-1/tab-2",
  "yes-version-yes-tabs/docs/api/version-1/tab-2/page-7",
  "yes-version-yes-tabs/docs/api/version-1/tab-2/page-8",
  "yes-version-yes-tabs/docs/api/version-1/tab-2/section-5",
  "yes-version-yes-tabs/docs/api/version-1/tab-2/section-5/page-10",
  "yes-version-yes-tabs/docs/api/version-2",
  "yes-version-yes-tabs/docs/api/version-2/tab-1",
  "yes-version-yes-tabs/docs/api/version-2/tab-1/section-1",
  "yes-version-yes-tabs/docs/api/version-2/tab-1/section-1/page-1",
  "yes-version-yes-tabs/docs/api/version-2/page-2",
  "yes-version-yes-tabs/docs/api/version-2/tab-1/section-2",
  "yes-version-yes-tabs/docs/api/version-2/section-2",
  "yes-version-yes-tabs/docs/api/version-2/tab-1/section-2/page-4",
  "yes-version-yes-tabs/docs/api/version-2/tab-1/page-5",
  "yes-version-yes-tabs/docs/api/version-2/tab-2",
  "yes-version-yes-tabs/docs/api/version-2/tab-2/page-7",
  "yes-version-yes-tabs/docs/api/version-2/tab-2/page-8",
  "yes-version-yes-tabs/docs/api/version-2/tab-2/section-5",
  "yes-version-yes-tabs/docs/api/version-2/tab-2/section-5/page-10",
]
`;

exports[`getNavigationRoot yes-version-yes-tabs gets navigation root for docs/api 1`] = `
{
  "redirect": "/docs/api/tab-1",
  "type": "redirect",
}
`;

exports[`getNavigationRoot yes-version-yes-tabs gets navigation root for docs/api/tab-1 1`] = `
{
  "found": {
    "currentNode": {
      "description": undefined,
      "fullSlug": [
        "tab-1",
      ],
      "id": "page-6.md",
      "slug": [
        "docs",
        "api",
        "tab-1",
      ],
      "title": "Page 6",
      "type": "page",
      "urlSlug": "page-6",
    },
    "currentTabIndex": 0,
    "currentVersionIndex": 0,
  },
  "type": "found",
}
`;

exports[`getNavigationRoot yes-version-yes-tabs gets navigation root for docs/api/tab-2 1`] = `
{
  "redirect": "/docs/api/tab-2/page-7",
  "type": "redirect",
}
`;

exports[`getNavigationRoot yes-version-yes-tabs gets navigation root for docs/api/version-1 1`] = `
{
  "redirect": "/docs/api/version-1/tab-1",
  "type": "redirect",
}
`;

exports[`getNavigationRoot yes-version-yes-tabs gets navigation root for docs/api/version-1/section-2 1`] = `
{
  "found": {
    "currentNode": {
      "description": undefined,
      "fullSlug": [
        "section-2",
      ],
      "id": "page-3.md",
      "slug": [
        "docs",
        "api",
        "version-1",
        "section-2",
      ],
      "title": "Page 3",
      "type": "page",
      "urlSlug": "page-3",
    },
    "currentTabIndex": 0,
    "currentVersionIndex": 1,
  },
  "type": "found",
}
`;

exports[`getNavigationRoot yes-version-yes-tabs gets navigation root for docs/api/version-2 1`] = `
{
  "redirect": "/docs/api/version-2/tab-1",
  "type": "redirect",
}
`;

exports[`getNavigationRoot yes-version-yes-tabs gets navigation root for docs/api/version-2/tab-2 1`] = `
{
  "redirect": "/docs/api/version-2/tab-2/page-7",
  "type": "redirect",
}
`;

exports[`getNavigationRoot yes-version-yes-tabs gets navigation root for docs/api/version-2/tab-2/section-5 1`] = `
{
  "redirect": "/docs/api/version-2/tab-1",
  "type": "redirect",
}
`;
